projectUtils/mockGame.js

```diff
@@ -56,6 +56,21 @@ class mockBC19 {
         }
     }
 
+    /**
+     * Method to take a BCAbstractRobot object and update the game's record of this bot with its communication information.
+     * Necessary because the game refers to a simplified `robot` object which does not reference `BCAbstractRobot.me` inherently
+     * and does not automatically adjust when bot adjusted.
+     * 
+     * @param {*} BCAbsBot BCAbstractRobot passed in to update the corresponding `robot` object in `this.game`
+     */
+    _setCommunication(BCAbsBot) {
```

Comment:
Can you explain this a little?

Reply:
Yeah so this is basically a necessary tweak to make sure any changes you make in `MyRobot` is reflected in the game. So say I want to set a specific castle talk value. I could do 
```const myBot = new MyRobot();
   myBot.castle_talk = 1;```
Without invoking this method, the other bots in the game won't see this value updated because of the design of `mockGame` and `game.js` in general. Calling this method ensures that another bot can see the `castle_talk` properly set to 1